cache.go

```diff
@@ -77,28 +77,50 @@ type Config struct {
 	// to the MaxAge
 	ExpirationInterval time.Duration
 	// Optional callback invoked when an item is evicted due to the LRU policy
-	OnEviction func(key, value interface{})
+	OnEviction func(key interface{}, value interface{})
 	// Optional callback invoked when an item expired
-	OnExpiration func(key, value interface{})
+	OnExpiration func(key interface{}, value interface{})
+}
+
+// ConfigGeneric configures the cache.
+type ConfigGeneric[T any] struct {
```

Comment:
I made a new struct just to make this backwards compatible.
The only thing that isn't backwards compatible, is that now the methods return pointers instead of just `interface{}`